src/metabase/integrations/ldap/default_implementation.clj

```diff
@@ -33,23 +32,34 @@
              :size-limit 1}))
           u/lower-case-map-keys))
 
+(s/defn ^:private process-group-membership-filter :- su/NonBlankString
+  "Replace DN and UID placeholders with values returned by the LDAP server."
+  [group-membership-filter :- su/NonBlankString
+   dn                      :- su/NonBlankString
+   uid                     :- (s/maybe su/NonBlankString)]
+  (let [uid-string (or uid "")]
```

Comment:
This seems awkward, is there any alternative? I don't think we can assume `uid` is set for a user, so we should default to an empty string if it's not, but doing this inline on L43 results in a reflection warning.

Reply:
If you put `^String` in front of the `(or ...)` it _should_ fix the reflection warning e.g.

```clj
(Filter/encodeValue ^String (or uid ""))
```

Reply:
That's what I had when I was seeing the reflection warning :/

Reply:
Can confirm, tried this again and the reflection linter prints:
```
Reflection warning, metabase/integrations/ldap/default_implementation.clj:46:30 - call to static method encodeValue on com.unboundid.ldap.sdk.Filter can't be resolved (argument types: unknown).
```